CHANGELOG.md

```diff
@@ -4,6 +4,9 @@
 
 ## Important Notes
 
+- [#849](https://github.com/oauth2-proxy/oauth2-proxy/pull/849) `/oauth2/auth` `allowed_groups` querystring parameter can be paired with the `allowed-groups` configuration option.
+  - In this scenario, the user's group must be in both lists to not get a 401 or 403 response code.
```

Comment:
Just to clarify, they must be in both because the provider checks the first one? There doesn't seem to be a crossover in the auth code here

Reply:
I suppose I should clarify `one of a user's groups must be present in both lists`?

Reply:
Yeah - the top `getAuthenticatedSession` has an embedded `Authorize` call which will do the normal global group authorization logic: https://github.com/grnhse/oauth2-proxy/blob/4e3aa3d6ade18af01b6b7df2c78c3cfedaebc4cc/oauthproxy.go#L1024

If that fails they get the 401 (because it likely means they are unauthenticated).

Then lower we have our follow up authorization check on querystring groups. That will return the 403, at this point we know they have a session, but they are forbidden, so no point in the 401 that would lead back to the sign in URL.

So layered gates where authorization must pass all of them.

Reply:
Scenario: I'm in groups `r&d` and `security`.  Global access proxy sets up access to anyone in `r&d`. I setup a security app via nginx ingress that I only want the `security` team to access, so it uses `?allowed_groups=security`.

My `r&d` group passes the first global authorization configuration gate. My second `security` group gets me past the app specific querystring version.

Then my SRE coworker in `r&d` and `sre` can't get into my security app.  But they can get into their Prometheus that might have `allowed_groups=sre` while I can't get in there.

We both can get into a generic engineering docs site that doesn't use the `allowed_groups` querystring since we are both in `r&d` for the global access proxy authorization rule.

Reply:
I added a detailed example in the Important Notes: https://github.com/oauth2-proxy/oauth2-proxy/pull/849/commits/c4df8649edb0eff1daaa96d620fdb2ae411e7daa